### DISCUSSION TITLE
How to cache Chocolately installs on GitHub Actions?

### DISCUSSION THREAD

**Discussion by lancejpollard** [chocolatey/choco#3388] [Q&A]
Cross-posting from https://stackoverflow.com/questions/77864421/how-to-cache-choco-downloads-in-github-actions

I have this in GitHub Actions:

[![enter image description here][1]][1]

From:

    on: push

    jobs:
      test:
        runs-on: windows-2022
        environment: build
        steps:
          - name: Checkout
            uses: actions/checkout@v3
            with:
              fetch-depth: 2
          - name: Create cache directory
            run: mkdir "$($env:LOCALAPPDATA)\choco-cache"
          - name: Cache Tools
            uses: actions/cache@v4
            with:
              path: "$($env:LOCALAPPDATA)\\choco-cache"
              key: ${{ runner.os }}-temp-cache
          - name: Set Choco cache path
            run: choco config set cacheLocation "$($env:LOCALAPPDATA)\choco-cache"
          - name: Install something
            run: choco install ffmpeg

It doesn't appear to do anything between runs, any ideas if I'm missing something or how to get this to work?

I see packages get installed in various places, like:

    Ruby is going to be installed in 'C:\tools\ruby33'
    ...
    pandoc installed to 'C:\Program Files\Pandoc'

Should I cache the `C:\Program Files` and `C:\tools` directories too or something?

From the little I saw on this, like https://github.com/chocolatey/choco/issues/1390, it appears it's not possible on GitHub actions? Or what do you think?

  [1]: https://i.stack.imgur.com/jzbnR.png

**Comment by TheCakeIsNaOH** [comment]
> Should I cache the C:\Program Files and C:\tools directories too or something?

Chocolatey packages generally will install software to the default location. So you would have to cache not only the Chocolatey package itself, but any and all locations that the software gets installed to. 

So instead of caching the installed package and it's software, I'd suggest caching the package file (the `.nupkg`) file. The Chocolatey Community Repository is not intended for direct organizational use, so an internal repository to cache the package is recommended. 
https://docs.chocolatey.org/en-us/community-repository/community-packages-disclaimer
https://docs.chocolatey.org/en-us/features/host-packages

There is a snippet here running on Appveyor which caches package files, which may be able to be adapted to Github actions.
https://github.com/chocolatey-community/chocolatey-packages/blob/1d43ae84611c24f478c692e904852701a3b4e1e5/.appveyor.yml#L59-L70
